Update installation documentation

Jasstkn opened the pull request:
Update documentation with this change https://github.com/litmuschaos/litmus-helm/pull/257.
Add more details about connecting via NodePort and ClusterIP

<!--  Thanks for sending a pull request!  Here are some tips for you -->

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
-   [ ] Fixes #<issue number>
-   [ ] Signed the commit for DCO to be passed
-   [ ] Labelled this PR & related issue with `documentation` tag

ksatchit commented on docs/openebs-control-plane-validation.md:
```diff
@@ -0,0 +1,217 @@
+
+## Prerequisites
+
+- Ensure that the Litmus Chaos Operator is running in the cluster. If not, install from [here](https://github.com/litmuschaos/chaos-operator/blob/master/deploy/operator.yaml)
```
Is this the link we are recommending or the pages one?

ksatchit commented on docs/openebs-control-plane-validation.md:
```diff
@@ -0,0 +1,217 @@
+- Ensure that the Litmus Chaos Operator is running in the cluster. If not, install from [here](https://github.com/litmuschaos/chaos-operator/blob/master/deploy/operator.yaml)
+- Ensure that the `openebs-control-plane-validation` experiment resource is available in the cluster. If not, install from [here](https://hub.litmuschaos.io/charts/openebs/experiments/openebs-control-plane-validation)
+
+- Ensure that the chaosServiceAccount used for the experiment has cluster-scope permissions as the experiment may involve carrying out the chaos in the `openebs` namespace
```
Did we use cluster scope rbac for this exp?

ksatchit commented on docs/openebs-control-plane-validation.md:
```diff
@@ -0,0 +1,217 @@
+
+## Details
+
+- This scenario validates the openEBS control plane forced termination of the control plane pod
```
Validates graceful & forced terminations of OpenEBS control plane pods

ksatchit commented on docs/openebs-control-plane-validation.md:
```diff
@@ -0,0 +1,217 @@
+
+## Integrations
+
+- Pod kill is achieved using the litmus and powerfullseal chaos library for docker runtime.
```
Powerfulseal

ksatchit replied:
Applies to other runtimes as well

ksatchit commented on docs/openebs-control-plane-validation.md:
```diff
@@ -0,0 +1,217 @@
+
+Use this sample RBAC manifest to create a chaosServiceAccount in the desired (app) namespace. This example consists of the minimum necessary cluster role permissions to execute the experiment.
+
+#### Sample Rbac Manifest
```
Can we use embedmd?

ksatchit commented on docs/openebs-control-plane-validation.md:
```diff
@@ -0,0 +1,217 @@
+    <th> Notes </th>
+  </tr>
+  <tr>
+    <td> APP_PVC </td>
```
Doesn't apply

ksatchit commented on docs/openebs-control-plane-validation.md:
```diff
@@ -0,0 +1,217 @@
+    <td> Defaults to `deployment`. Supported: `deployment`, `statefulset` </td>
+  </tr>
+  <tr>
+    <td> DATA_PERSISTENCE </td>
```
Doesn't apply

ksatchit commented on docs/openebs-control-plane-validation.md:
```diff
@@ -0,0 +1,217 @@
+    <td> Defaults to `gaiaadm/pumba:0.4.8`. Supported: `gaiaadm/pumba:0.4.8` </td>
+  </tr>
+  <tr>
+    <td> DEPLOY_TYPE </td>
```
Aren't we taking this in appinfo

ksatchit commented on docs/openebs-control-plane-validation.md:
```diff
@@ -0,0 +1,217 @@
+
+### Watch Chaos progress
+
+- View pod restart count by setting up a watch on the pods in the OpenEBS namespace
```
Restart counts are only for container kills right

ksatchit commented on docs/openebs-control-plane-validation.md:
```diff
@@ -0,0 +1,217 @@
+- Check whether the openEBS control plane is resilient to the pod failure, once the experiment (job) is completed. The ChaosResult resource naming convention 
+  is: `<ChaosEngine-Name>-<ChaosExperiment-Name>`.
+
+  `kubectl describe chaosresult target-chaos-openebs-pool-container-failure -n <application-namespace>`
```
This is control plane validation right

shubhamchaudhary replied:
also, add `openebs` ns instead of `<application-namespace>``

shubhamchaudhary commented on docs/openebs-control-plane-validation.md:
```diff
@@ -0,0 +1,196 @@
+## Integrations
+
+- Pod kill is achieved using the litmus or powerfulseal chaos libraries.
+- The desired lib can be configured in the env variable `LIB`.
```
Can you pls mention the `litmus` & `powerfulseal` as supported fields

shubhamchaudhary commented on docs/openebs-control-plane-validation.md:
```diff
@@ -0,0 +1,196 @@
+
+### Prepare chaosServiceAccount
+
+Use this sample RBAC manifest to create a chaosServiceAccount in the desired (app) namespace. This example consists of the minimum necessary role permissions to execute the experiment.
```
Use this sample RBAC manifest to create a chaosServiceAccount in the desired (openebs) namespace. This example consists of the minimum necessary role permissions to execute the experiment.

shubhamchaudhary commented on docs/openebs-control-plane-validation.md:
```diff
@@ -0,0 +1,196 @@
+    <th> Type </th>
+    <th> Notes </th>
+  </tr>
+  <tr>
```
<tr> is occuring twice here

shubhamchaudhary commented on docs/openebs-control-plane-validation.md:
```diff
@@ -0,0 +1,196 @@
+  </tr>
+  <tr>
+  <tr>
+    <td> LIB_IMAGE </td>
```
This experiments support `LIB_IMAGE `? and pls add `LIB`

shubhamchaudhary commented on docs/openebs-control-plane-validation.md:
```diff
@@ -0,0 +1,196 @@
+
+- Create the ChaosEngine manifest prepared in the previous step to trigger the Chaos.
+
+  `kubectl apply -f chaosengine.yml`
```
pls mention namespace here `openebs`

shubhamchaudhary commented on docs/openebs-control-plane-validation.md:
```diff
@@ -0,0 +1,196 @@
+
+- View pod restart by setting up a watch on the pods in the OpenEBS namespace
+
+  `watch -n 1 kubectl get pods -n <openebs-namespace>`
```
`watch -n 1 kubectl get pods -n openebs

shubhamchaudhary commented on docs/openebs-control-plane-validation.md:
```diff
@@ -0,0 +1,196 @@
+### Check Chaos Experiment Result
+
+- Check whether the openEBS control plane is resilient to the pod failure, once the experiment (job) is completed. The ChaosResult resource naming convention 
+  is: `<ChaosEngine-Name>-<ChaosExperiment-Name>`.
```
why you added this in new line , can you add this in last line itself

shubhamchaudhary commented on docs/openebs-control-plane-validation.md:
```diff
@@ -0,0 +1,196 @@
+
+  `kubectl describe chaosresult target-chaos-openebs-pool-container-failure -n <application-namespace>`
+
+## OpenEBS Pool Container Failure Demo [TODO]
```
OpenEBS Control Plane Validation Demo [TODO]

shubhamchaudhary commented on docs/openebs-pool-network-delay.md:
```diff
@@ -0,0 +1,259 @@
+## Prerequisites
+
+- Ensure that the Kubernetes Cluster uses Docker runtime
+- Ensure that the Litmus Chaos Operator is running in the cluster. If not, install from [here](https://raw.githubusercontent.com/litmuschaos/pages/master/docs/litmus-operator-latest.yaml)
```
Ensure that the Litmus Chaos Operator is running by executing `kubectl get pods` in operator namespace (typically, `litmus`). If not, install from [here](https://raw.githubusercontent.com/litmuschaos/pages/master/docs/litmus-operator-latest.yaml)

shubhamchaudhary commented on docs/openebs-pool-network-delay.md:
```diff
@@ -0,0 +1,259 @@
+
+- Ensure that the Kubernetes Cluster uses Docker runtime
+- Ensure that the Litmus Chaos Operator is running in the cluster. If not, install from [here](https://raw.githubusercontent.com/litmuschaos/pages/master/docs/litmus-operator-latest.yaml)
+- Ensure that the `openebs-pool-network-delay` experiment resource is available in the cluster. If not, install from [here](https://hub.litmuschaos.io/charts/openebs/experiments/openebs-pool-network-delay)
```
pls add kubectl command to check the experiment

shubhamchaudhary commented on docs/openebs-target-network-loss.md:
```diff
@@ -240,7 +240,7 @@ spec:
 - Watch the behaviour of the application pod and the OpenEBS data replica/pool pods by setting up in a watch on the respective namespaces
 
-  `watch -n 1 kubectl get pods -n <app/openebs-namespace>`
+  `watch -n 1 kubectl get pods -n <openebs-namespace>`
```
Its should be app-namespace for all experiment except `openebs-control-plane-validation`

Jasstkn committed:
Update installation documentation

Update documentation with this change https://github.com/litmuschaos/litmus-helm/pull/257.
Add more details about connecting via NodePort and ClusterIP

Signed-off-by: Jasstkn <mariia.kotliarevskaia@gmail.com>

Jasstkn commented:
@divya-mohan0209 @ksatchit @k8s-dev could you check this PR? It should be merged after the one that is mentioned in the description.

Jonsy13 commented:
Hi @Jasstkn Changes looks good, Can you update these docs also
- (Master docs) -  https://github.com/litmuschaos/litmus-docs/blob/master/website/docs/getting-started/installation.md
- (2.12.0 docs, since we are updating helm-chart for same version) - https://github.com/litmuschaos/litmus-docs/blob/master/website/versioned_docs/version-2.12.0/getting-started/installation.md

Jasstkn committed:
Merge remote-tracking branch 'origin/master' into patch-1

Jasstkn committed:
Fix docs

Signed-off-by: Maria Kotlyarevskaya <mariia.kotliarevskaia@gmail.com>
